src/server/Session.ts

```diff
@@ -13,8 +14,11 @@ export default class Session {
 
   private _state: SessionState = SessionState.Waiting;
 
-  constructor() {
+  private _settings: SessionSettings;
+
+  constructor(settings: SessionSettings) {
```

Comment:
You can use typescript's shorthand for this.
```suggestion
  constructor(private _settings: SessionSettings) {
```

Also, I forget why these are prefixed with `_` 🤔